File: hackerrank/birthdayCakeCandles.py
# https://www.hackerrank.com/challenges/birthday-cake-candles/problem

def birthdayCakeCandles(candles: 'list[int]') -> int:
    tallestNumber = float('-inf')
    tallestCount = 0

    for x in candles:
        if x > tallestNumber:
            tallestNumber = x
            tallestCount = 1
        elif x == tallestCount:
            tallestCount += 1

    return tallestCount

candlesExample = [3, 2, 1, 3]
print(birthdayCakeCandles(candlesExample)) # output 2

File: hackerrank/birthdayCakeCandles.js
// https://www.hackerrank.com/challenges/birthday-cake-candles/problem
/**
 *
 * @param {Array<number>} candles
 * @returns {number}
 */

function birthdayCakeCandles(candles) {
  const tallestNumber = -Infinity;
  const tallestCount = 0;

  for(const candle in candles) {
    switch(true) {
      case candle > tallestNumber:
        tallestNumber = candle;
        tallestCount = 1;
        break;
      case candle === tallestNumber:
        tallestCount += 1;
        break;
      default:
        continue;
    }
  }

  return tallestCount;
}

const candlesExample = [3, 2, 1, 3];
console.log(birthdayCakeCandles(candlesExample)); // output: 2
